notifier/manager_test.go

```diff
@@ -191,34 +196,66 @@ func TestHandlerSendAll(t *testing.T) {
 		}
 	}
 
-	// all ams in all sets are up
-	require.True(t, h.sendAll(h.queue...), "all sends failed unexpectedly")
+	// start send loops
+	for _, ams := range h.alertmanagers {
+		ams.startSendLoops(ams.ams)
+	}
+
+	h.Send(alerts...)
+	// wait for metrics to update
+	time.Sleep(time.Second)
+	require.Zero(t, prom_testutil.ToFloat64(h.metrics.errors.WithLabelValues(server1.URL)))
+	require.Zero(t, prom_testutil.ToFloat64(h.metrics.errors.WithLabelValues(server2.URL)))
+	require.Zero(t, prom_testutil.ToFloat64(h.metrics.errors.WithLabelValues(server3.URL)))
 	checkNoErr()
 
 	// the only am in set 1 is down
 	status1.Store(int32(http.StatusNotFound))
-	require.False(t, h.sendAll(h.queue...), "all sends failed unexpectedly")
+	h.Send(alerts...)
+	// wait for metrics to update
+	time.Sleep(time.Second)
+	require.NotZero(t, prom_testutil.ToFloat64(h.metrics.errors.WithLabelValues(server1.URL)))
+	require.Zero(t, prom_testutil.ToFloat64(h.metrics.errors.WithLabelValues(server2.URL)))
+	require.Zero(t, prom_testutil.ToFloat64(h.metrics.errors.WithLabelValues(server3.URL)))
 	checkNoErr()
 
-	// reset it
+	// fix the am
 	status1.Store(int32(http.StatusOK))
 
+	// reset metrics
+	h.metrics.errors.Reset()
+
 	// only one of the ams in set 2 is down
 	status2.Store(int32(http.StatusInternalServerError))
-	require.True(t, h.sendAll(h.queue...), "all sends succeeded unexpectedly")
+	h.Send(alerts...)
+	// wait for metrics to update
+	time.Sleep(time.Second)
+	require.Zero(t, prom_testutil.ToFloat64(h.metrics.errors.WithLabelValues(server1.URL)))
+	require.NotZero(t, prom_testutil.ToFloat64(h.metrics.errors.WithLabelValues(server2.URL)))
+	require.Zero(t, prom_testutil.ToFloat64(h.metrics.errors.WithLabelValues(server3.URL)))
 	checkNoErr()
 
 	// both ams in set 2 are down
 	status3.Store(int32(http.StatusInternalServerError))
-	require.False(t, h.sendAll(h.queue...), "all sends succeeded unexpectedly")
+	h.Send(alerts...)
+	// wait for metrics to update
+	time.Sleep(time.Second)
+	require.Zero(t, prom_testutil.ToFloat64(h.metrics.errors.WithLabelValues(server1.URL)))
+	require.NotZero(t, prom_testutil.ToFloat64(h.metrics.errors.WithLabelValues(server2.URL)))
+	require.NotZero(t, prom_testutil.ToFloat64(h.metrics.errors.WithLabelValues(server3.URL)))
 	checkNoErr()
+
+	// stop send loops.
+	for _, ams := range h.alertmanagers {
+		ams.cleanSendLoops(ams.ams)
+	}
```

Comment:
nit: should be in a defered function defined after the loops are started . (same for later usages)